### PR TITLE
Document new text index provider and deprecation of the old one

### PR DESCRIPTION
**Pull request by Hunterness**

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 1]
```diff
@@ -209,7 +222,7 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
       title = "Create a text index specifying the index provider",
       text =
         s"""To create a text index with a specific index provider, the `OPTIONS` clause is used.
-          |Only one valid value exists for the index provider, `$textProvider`, which is the default value.""".stripMargin,
+          |The valid values for the index provider is `$textProvider` and `$oldTextProvider` (deprecated), the default value is $textProvider.""".stripMargin,
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]:
```suggestion
          |The valid values for the index provider are `$textProvider` and `$oldTextProvider` (deprecated). The default provider is $textProvider.""".stripMargin,
```

**Review by mnd999 — CHANGES_REQUESTED** [review]

**Review by mnd999 — APPROVED** [review]

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -10,6 +10,49 @@ New features are added to the language continuously, and occasionally, some feat
...
+a|
+label:functionality[]
+label:updated[]
+[source, cypher, role="noheader"]
+----
+CREATE TEXT INDEX ... OPTIONS {indexProvider: `text-2.0`}
+----
+a|
+
+A new text index provider is available, `text-2.0`. This is also the default provider if none is given.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
A new text index provider is available, `text-2.0`. 
This is also the default provider if none is given.
```

**Review by AlexicaWright — COMMENTED** [review]
One small comment only.

**Review by AlexicaWright — APPROVED** [review]
👍

**Pull request merged**